osu.Game/Screens/OsuScreen.cs

```diff
@@ -140,6 +146,18 @@ private void load(OsuGame osu, AudioManager audio)
             Activity.Value ??= InitialActivity;
         }
 
+        /// <summary>
+        /// Apply arbitrary changes to the current background screen in a thread safe manner.
+        /// </summary>
+        /// <param name="action">The operation to perform.</param>
+        public void ApplyToBackground(Action<BackgroundScreen> action)
+        {
+            if (background == null)
+                throw new InvalidOperationException("Attempted to apply to background before screen is pushed");
```

Comment:
Hmm, this can actually surface in a different way, namely - `backgroundStack` is allowed to be null, so even after push this could potentially be triggered. Maybe we should be distinguishing between the two cases by explicitly checking?

Reply:
Very edge case, but I've split this out (i guess it'd only happen in tests).